fragment/fragment-lint/src/main/java/androidx/fragment/lint/UseGetLayoutInflater.kt

```diff
@@ -0,0 +1,130 @@
+import com.android.tools.lint.detector.api.LintFix
+import com.android.tools.lint.detector.api.Scope
+import com.android.tools.lint.detector.api.Severity
+import com.android.tools.lint.detector.api.SourceCodeScanner
+import com.android.tools.lint.detector.api.isKotlin
+import com.intellij.psi.PsiMethod
+import org.jetbrains.uast.UCallExpression
+import org.jetbrains.uast.getContainingUClass
+
+/**
+ * Lint check for detecting calls to [android.view.LayoutInflater.from]
+ * while being invoked from DialogFragment
+ */
+class UseGetLayoutInflater : Detector(), SourceCodeScanner {
+
+    override fun getApplicableMethodNames(): List<String>? {
+        return listOf(UNWANTED_METHOD)
+    }
+
+    override fun visitMethodCall(context: JavaContext, node: UCallExpression, method: PsiMethod) {
+        val containingClass = method.containingClass ?: return
+        val evaluator = context.evaluator
+        if (evaluator.getQualifiedName(containingClass) == UNWANTED_CLASS &&
+            evaluator.getParameterCount(method) == 1
+        ) {
+            if (!isKotlin(context.psiFile)) {
+                startLintForJava(context, node)
+            } else {
+                startLintForKotlin(context, node)
+            }
+        }
+    }
+
+    private fun startLintForJava(context: JavaContext, node: UCallExpression) {
+        if (node.getContainingUClass()?.superClassType?.name != DIALOG_FRAGMENT_CLASS) {
+            return
+        }
+        val methodParameter = node.valueArguments[0].toString()
+
+        context.report(
+            issue = ISSUE,
+            location = context.getLocation(node),
+            message = "Use of LayoutInflater.from($methodParameter) detected. Consider using " +
+                "${correctMethod(context)} instead",
+            quickfixData = createFix(correctMethod(context), methodParameter)
+        )
+    }
+
+    private fun startLintForKotlin(context: JavaContext, node: UCallExpression) {
+        if (node.getContainingUClass()?.javaPsi?.text?.contains
+                ("${DIALOG_FRAGMENT_CLASS}()") == false
+        ) {
+            return
+        }
+
+        context.report(
+            issue = ISSUE,
+            location = context.getLocation(node),
+            message = "Use of LayoutInflater.from(Context) detected. Consider using " +
+                "${correctMethod(context)} instead",
+            quickfixData = null
+        )
+    }
+
+    private fun correctMethod(context: JavaContext): String {
+        return if (isKotlin(context.psiFile)) {
+            "layoutInflater"
+        } else {
+            "getLayoutInflater()"
+        }
+    }
+
+    private fun createFix(correctMethod: String, parameter: String?): LintFix {
+        return fix()
+            .replace()
+            .text("LayoutInflater.from($parameter)")
+            .name("Replace with $correctMethod")
+            .with(correctMethod)
+            .autoFix()
+            .build()
+    }
+
+
+    companion object Issues {
```

Comment:
Place the companion object at the top of the file to be consistent with our current lint rules.